### DISCUSSION TITLE
eventTargetOf  -- 是怎么获取所作用到的元素的

### DISCUSSION THREAD

**Discussion by wuyz-harder** [antvis/G6#5743] [Q&A]
大佬们antv这个库写的太好了，勾起了我看源码的好奇心。有个问题想请教一下，我在看g6   5.0.0-beta.35版本代码的时候；从runtime/behavior.ts下的 forwardCanvasEvents -》 eventTargetOf  的代码里，我没有找到哪部分的逻辑是canvas的事件转化到比如说node\edge上的？正常来说，应该是有个事件触发的clientX/Y => pointX/Y的转换，然后计算所有元素，找出所作用的节点的吧？大佬,sos,想知道一下这部分逻辑是哪部分代码处理的

**Comment by Aarebecca** [comment]
你指的这部分逻辑是由底层渲染引擎完成的，eventTargetOf 仅仅是遍历父节点，并找到最近的元素节点。
你可以参考：https://g.antv.antgroup.com/api/builtin-objects/event-target
